### PR TITLE
Issue 3124 - make 'glooctl debug logs -f' actually save to file

### PR DESCRIPTION
**Pull request by mlholland**
# Description

This fixes a small bug in the glooctl debug command, in which the logs sub-command only respected the -f flag when --zip was also included. The command's functionality now matches the behavior specified in the docs by outputting logs to a text file when -f <filename> is included without --zip.

# Context

See https://github.com/solo-io/gloo/issues/3124

# Checklist:

- [X] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [X] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [X] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3124

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/debug/debug_test.go [hunk 1]
```diff
@@ -50,6 +50,21 @@ var _ = Describe("Debug", func() {
 			err = os.RemoveAll(opts.Top.File)
 			Expect(err).NotTo(HaveOccurred())
 		})
+
+		It("should create a text file at location specified in --file when --zip is not enabled", func() {
+			opts := options.Options{}
+			opts.Metadata.Namespace = "gloo-system"
+			opts.Top.File = "/tmp/log.txt"
```
Comment (GrahamGoudeau) [inline projects/gloo/cli/pkg/cmd/debug/debug_test.go:57]: I think you want https://golang.org/pkg/io/ioutil/#TempDir instead

Comment (mlholland) [reply projects/gloo/cli/pkg/cmd/debug/debug_test.go]: Will do, changing both this and the original zip test that this code came from

**Comment by kdorosh** [comment]
/kick known regression test flake, https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=aedc4197-3727-4ba6-a917-8ae5683ec19c as reported here https://github.com/solo-io/gloo/issues/3226

File: projects/gloo/cli/pkg/cmd/debug/debug.go [hunk 2]
```diff
@@ -41,6 +41,12 @@ func DebugLogs(opts *options.Options, w io.Writer) error {
 	defer fs.RemoveAll(dir)
 	storageClient := debugutils.NewFileStorageClient(fs)
 
+	// if writing to a non-zipped file, create a channel to collect logs
+	var fileBuf chan string
+	if !opts.Top.Zip && opts.Top.File != "" {
+		fileBuf = make(chan string, len(responses))
```
Comment (GrahamGoudeau) [inline projects/gloo/cli/pkg/cmd/debug/debug.go:47]: Can we `defer func() { close(fileBuf) }` here?

Comment (mlholland) [reply projects/gloo/cli/pkg/cmd/debug/debug.go:47]: We already close it (without a defer) once just before we need to read its inputs below, and according to google, double closing a channel causes a panic.

**Pull request merged**